src/Components/CustomElements/src/wwwroot/Microsoft.AspNetCore.Components.CustomElements.lib.module.js

```diff
@@ -0,0 +1 @@
+import './BlazorCustomElements.js';
```

Comment:
As an alternative to adding an extra file, we could do [this approach](https://github.com/dotnet/aspnetcore/commit/39a3771aed5bb71d8d60840e16dbe83ab0202c30) where the webpack-generated JS file serves as the JS initializer (this would get rid of the additional JS file). But the main advantage I see of keeping it as-is would be that customers who have already started using this feature won't get 404s when loading a nonexistent script.

And, while it works, I'm not sure how valid the `SourceType="Discovered"` hack is (in the commit linked above).

Thoughts?

Reply:
Personally I'd bias towards making the product optimal in the long run over retaining back-compat with a preview release. This is the first time custom elements is becoming fully-supported, so it's our last easy chance to make any breaking changes that will improve things long-term.

> And, while it works, I'm not sure how valid the SourceType="Discovered" hack is (in the commit linked above).

Fully agreed that this is super-low-discoverability, but TBH I've never managed to remember these kinds of details about configuring SWA, so for me at least `SourceType="Computed"` is equally nonobvious. Is there some reason to regard `Discovered` as a "hack" or is this meant to be supported and is it the correct approach, @javiercn?

Reply:
I'm with @SteveSandersonMS here, let's make the webpack output the initializer itself to avoid the extra file. Even with HTTP (2/3) those still matter.

Reply:
> Is there some reason to regard `Discovered` as a "hack" (as opposed to `Computed`) or is this meant to be supported and is it the correct approach, @javiercn?

It's very likely just because it is looking only at the discovered items [here](https://github.com/dotnet/sdk/blob/main/src/RazorSdk/Targets/Microsoft.NET.Sdk.Razor.JSModules.targets#L69).

Longer term we need to make the decision (as part of client assets) of whether we want the generated assets to be defined as assets or as Content items. My preference for now is that the output of that process is that the generated elements end up defined as Content items because that allows the rest of the pipeline to work without changes.